packages/grid/_modules_/grid/components/cell/GridEditInputCell.tsx

```diff
@@ -33,13 +38,15 @@ export function GridEditInputCell(props: GridCellParams & InputBaseProps) {
       };
 
       if (isDateColumn) {
-        editProps.value = newValue === '' ? null : new Date(newValue); // TODO fix parsing, this is plain wrong.
+        editProps.value = newValue === '' ? null : formatDateString(newValue);
+      } else if (isDateTimeColumn) {
+        editProps.value = newValue === '' ? null : new Date(newValue);
```

Comment:
We can't do it, we support Safari 12.1: https://github.com/mui-org/material-ui/pull/23475#issuecomment-839640332

Reply:
When editing a date or dateTime cell in Safari, any changes to the cell cause "Invalid Date" to render. The same occurs in Firefox for dateTime cells. Should this be a separate issue/PR?

Reply:
Is the PR rebased on HEAD? We have worked on a similar issue recently, on Chrome.

Reply:
Otherwise we could leave a to-do, we already improve one aspect.

Reply:
> Is the PR rebased on HEAD? We have worked on a similar issue recently, on Chrome.

Yes, this is happening in the current version of the docs

Reply:
Ok, so I had a look:

- Firefox. Doesn't support `<input type="datetime-local">`
- Safari. I can't reproduce the issue, however, the onChange event doesn't seem to fire reliably once emptied, and the text vertical alignment is wrong.

It seems that we will need to drop the native implementation.

Reply:
Here's a GIF of this happening in Safari 14.0.3

![Screen Capture](https://user-images.githubusercontent.com/37122448/118165299-c0e64380-b3e9-11eb-8559-21fbfed52c54.gif)

Reply:
I'm having trouble coming up with a solution for this. I'm guessing the preferred behavior would be to :

1. Allow an invalid date while a cell is being edited
2. Render "Invalid Date" upon a submit event/enter/etc. if the value is not a date.

I'm unable to get both 1. and 2. because there is no `onSubmit` prop with `InputBase`. And the function passed to `onChange` does not run on <kbd>Enter</kbd>.

Is there a different way I should go about this?

Reply:
@aTmb405 I would give up on it. Native date inputs have limitations and unlikely to be something developers can rely on.